### PR TITLE
LIMS-6853: delete suborder

### PR DESCRIPTION
**Pull request by ghada-37**
**In case I have main order with multiple sub orders, make sure you can delete one of them**
```
➜  1lims-automation git:(LIMS-6853) ✗ nosetests -vs --nologcapture --tc-file=config.ini ui_testing/testcases/basic_tests/test06_orders.py:OrdersTestCases.test033_delete_suborder
2020-04-08 22:33:49.779 | INFO     | api_testing.apis.base_api:info:70 - Get authorized api session.
2020-04-08 22:33:50.582 | INFO     | api_testing.apis.base_api:info:70 - session ID : eyJhbGciOi .....
Delete sub order Approach: In case I have main order with multiple sub orders, ...      
2020-04-08 22:33:50.746 | INFO     | ui_testing.testcases.base_test:info:122 - Test case : test033_delete_suborder
2020-04-08 22:34:42.105 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:523 - wait until page is loaded
2020-04-08 22:34:42.105 | INFO     | ui_testing.pages.base_selenium:wait_until_element_is_not_displayed:187 - wait until element is not displayed
2020-04-08 22:34:43.315 | INFO     | ui_testing.testcases.base_test:info:122 - get random order with multiple suborders
2020-04-08 22:34:43.316 | INFO     | api_testing.apis.base_api:info:70 - GET : https://automation.1lims.com/api/orders
2020-04-08 22:34:44.205 | INFO     | api_testing.apis.base_api:info:70 - Status code: 1
2020-04-08 22:34:44.206 | INFO     | api_testing.apis.base_api:info:70 - GET : https://automation.1lims.com/api/orders/get/subOrder/byMainOrderId?orderId=987&deleted=0
2020-04-08 22:34:44.602 | INFO     | api_testing.apis.base_api:info:70 - Status code: 1
2020-04-08 22:34:44.602 | INFO     | api_testing.apis.base_api:info:70 - GET : https://automation.1lims.com/api/orders/get/subOrder/byMainOrderId?orderId=985&deleted=0
2020-04-08 22:34:44.954 | INFO     | api_testing.apis.base_api:info:70 - Status code: 1
2020-04-08 22:34:44.954 | INFO     | api_testing.apis.base_api:info:70 - GET : https://automation.1lims.com/api/orders/get/subOrder/byMainOrderId?orderId=984&deleted=0
2020-04-08 22:34:45.345 | INFO     | api_testing.apis.base_api:info:70 - Status code: 1
2020-04-08 22:34:45.345 | INFO     | api_testing.apis.base_api:info:70 - GET : https://automation.1lims.com/api/orders/get/subOrder/byMainOrderId?orderId=983&deleted=0
2020-04-08 22:34:45.759 | INFO     | api_testing.apis.base_api:info:70 - Status code: 1
2020-04-08 22:34:45.759 | INFO     | api_testing.apis.base_api:info:70 - GET : https://automation.1lims.com/api/orders/get/subOrder/byMainOrderId?orderId=980&deleted=0
2020-04-08 22:34:46.110 | INFO     | api_testing.apis.base_api:info:70 - Status code: 1
2020-04-08 22:34:46.110 | INFO     | api_testing.apis.base_api:info:70 - GET : https://automation.1lims.com/api/orders/get/subOrder/byMainOrderId?orderId=979&deleted=0
2020-04-08 22:34:46.475 | INFO     | api_testing.apis.base_api:info:70 - Status code: 1
2020-04-08 22:34:46.475 | INFO     | api_testing.apis.base_api:info:70 - GET : https://automation.1lims.com/api/orders/get/subOrder/byMainOrderId?orderId=978&deleted=0
2020-04-08 22:34:46.884 | INFO     | api_testing.apis.base_api:info:70 - Status code: 1
2020-04-08 22:34:46.884 | INFO     | api_testing.apis.base_api:info:70 - GET : https://automation.1lims.com/api/orders/get/subOrder/byMainOrderId?orderId=977&deleted=0
2020-04-08 22:34:47.235 | INFO     | api_testing.apis.base_api:info:70 - Status code: 1
2020-04-08 22:34:47.236 | INFO     | api_testing.apis.base_api:info:70 - GET : https://automation.1lims.com/api/orders/get/subOrder/byMainOrderId?orderId=976&deleted=0
2020-04-08 22:34:47.605 | INFO     | api_testing.apis.base_api:info:70 - Status code: 1
2020-04-08 22:34:47.606 | INFO     | api_testing.apis.base_api:info:70 - GET : https://automation.1lims.com/api/orders/get/subOrder/byMainOrderId?orderId=975&deleted=0
2020-04-08 22:34:48.012 | INFO     | api_testing.apis.base_api:info:70 - Status code: 1
2020-04-08 22:34:48.013 | INFO     | api_testing.apis.base_api:info:70 - GET : https://automation.1lims.com/api/orders/get/subOrder/byMainOrderId?orderId=974&deleted=0
2020-04-08 22:34:48.356 | INFO     | api_testing.apis.base_api:info:70 - Status code: 1
2020-04-08 22:34:48.356 | INFO     | api_testing.apis.base_api:info:70 - GET : https://automation.1lims.com/api/orders/get/subOrder/byMainOrderId?orderId=973&deleted=0
2020-04-08 22:34:48.727 | INFO     | api_testing.apis.base_api:info:70 - Status code: 1
2020-04-08 22:34:48.728 | INFO     | api_testing.apis.base_api:info:70 - GET : https://automation.1lims.com/api/orders/get/subOrder/byMainOrderId?orderId=972&deleted=0
2020-04-08 22:34:49.137 | INFO     | api_testing.apis.base_api:info:70 - Status code: 1
2020-04-08 22:34:49.765 | INFO     | ui_testing.pages.base_selenium:wait_until_element_clickable:206 - wait until element clickable
2020-04-08 22:35:00.043 | INFO     | ui_testing.testcases.base_test:info:122 - archive first suborder
2020-04-08 22:35:00.272 | INFO     | ui_testing.pages.base_pages:sleep_medium:47 -  Medium sleep.
2020-04-08 22:35:17.969 | INFO     | ui_testing.pages.base_pages:info:271 - archive suborder from the order's active table
2020-04-08 22:35:18.014 | INFO     | ui_testing.pages.base_pages:info:271 - open record options menu
2020-04-08 22:35:18.197 | INFO     | ui_testing.pages.base_pages:sleep_tiny:39 -  Tiny sleep.
2020-04-08 22:35:20.200 | INFO     | ui_testing.pages.base_selenium:wait_until_element_clickable:206 - wait until element clickable
2020-04-08 22:35:20.413 | INFO     | ui_testing.pages.base_pages:confirm_popup:73 - Confirming the popup
2020-04-08 22:35:20.461 | INFO     | ui_testing.pages.base_selenium:wait_until_element_clickable:206 - wait until element clickable
2020-04-08 22:35:30.722 | INFO     | ui_testing.pages.base_pages:info:271 - navigate to archived order table
2020-04-08 22:35:30.733 | INFO     | ui_testing.pages.base_selenium:wait_until_element_clickable:206 - wait until element clickable
2020-04-08 22:35:30.931 | INFO     | ui_testing.pages.base_selenium:wait_until_element_clickable:206 - wait until element clickable
2020-04-08 22:35:31.421 | INFO     | ui_testing.pages.base_pages:sleep_small:43 -  Small sleep.
2020-04-08 22:35:40.943 | INFO     | ui_testing.pages.base_pages:info:271 - delete suborder with index 1 and analysis_no 1'470-20
2020-04-08 22:35:40.944 | INFO     | ui_testing.pages.base_pages:info:271 - open record options menu
2020-04-08 22:35:41.109 | INFO     | ui_testing.pages.base_pages:sleep_tiny:39 -  Tiny sleep.
2020-04-08 22:35:43.111 | INFO     | ui_testing.pages.base_selenium:wait_until_element_clickable:206 - wait until element clickable
2020-04-08 22:35:43.330 | INFO     | ui_testing.pages.base_pages:confirm_popup:73 - Confirming the popup
2020-04-08 22:35:43.382 | INFO     | ui_testing.pages.base_selenium:wait_until_element_clickable:206 - wait until element clickable
2020-04-08 22:35:53.640 | INFO     | ui_testing.pages.base_pages:confirm_popup:73 - Confirming the popup
2020-04-08 22:35:53.671 | INFO     | ui_testing.pages.base_selenium:wait_until_element_clickable:206 - wait until element clickable
2020-04-08 22:36:03.900 | INFO     | ui_testing.testcases.base_test:info:122 - Navigate to order page to make sure that suborder is deleted and main order still active
2020-04-08 22:36:07.793 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:523 - wait until page is loaded
2020-04-08 22:36:07.794 | INFO     | ui_testing.pages.base_selenium:wait_until_element_is_not_displayed:187 - wait until element is not displayed
2020-04-08 22:36:11.449 | INFO     | ui_testing.pages.base_selenium:wait_until_element_clickable:206 - wait until element clickable
2020-04-08 22:36:21.988 | INFO     | ui_testing.pages.base_pages:sleep_medium:47 -  Medium sleep.
2020-04-08 22:36:37.995 | INFO     | ui_testing.testcases.base_test:info:122 - Navigate to Analysis page to make sure that analysis related to deleted suborder not found
2020-04-08 22:36:38.003 | INFO     | ui_testing.pages.base_selenium:wait_until_element_clickable:206 - wait until element clickable
2020-04-08 22:36:38.434 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:523 - wait until page is loaded
2020-04-08 22:36:38.434 | INFO     | ui_testing.pages.base_selenium:wait_until_element_is_not_displayed:187 - wait until element is not displayed
2020-04-08 22:36:41.270 | INFO     | ui_testing.pages.base_pages:open_filter_menu:82 -  Open Filter
2020-04-08 22:36:42.406 | INFO     | ui_testing.pages.base_selenium:wait_until_element_clickable:206 - wait until element clickable
2020-04-08 22:36:47.688 | INFO     | ui_testing.pages.base_pages:sleep_tiny:39 -  Tiny sleep.
2020-04-08 22:36:49.823 | INFO     | ui_testing.testcases.base_test:info:122 - TearDown.        
ok

----------------------------------------------------------------------
Ran 1 test in 179.078s

OK


```

### CODE REVIEW

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 1]
```diff
@@ -2118,3 +2118,34 @@ def test033_Duplicate_main_order_and_cahange_materiel_type(self):
...
+        self.order_page.get_archived_items()
+        row = self.order_page.search(order['orderNo'])
+        self.order_page.click_check_box(row[0])
+        self.order_page.delete_selected_item()
+        validation_result = self.base_selenium.get_text(element='orders:confirm_delete')
+        self.assertIn('Please note that the reports of this order will be deleted', validation_result)
+        self.order_page.confirm_popup()
+        # go to analysis page to make sure that analysis reated to that suborder is deleted
+        self.info('Navigate to analysis page to make sure that analysis related to that suborder is deleted')
+        self.order_page.get_orders_page()
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test06_orders.py:2145]: after it, assert that the main order still existing with the other sub orders (step-7)

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 2]
```diff
@@ -2118,3 +2118,34 @@ def test033_Duplicate_main_order_and_cahange_materiel_type(self):
...
         self.assertEqual(suborder_data['Test Units'], test_unit)
+
+    def test033_delete_suborder(self):
+        """
+         Delete sub order Approach: In case I have main order with multiple sub orders ,
+         make sure you can delete one of them
+
+        'LIMS-6853'
+        """
+        # get the random main order data
```
[the remark this answers is not in the document]
Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test06_orders.py:2129]: done

File: api_testing/apis/orders_api.py [hunk 3]
```diff
@@ -213,3 +213,13 @@ def _format_payload(payload):
 
         payload['materialTypeId'] = payload['materialType']['id']
         return payload
+
+class  OrdersAPI(OrdersAPIFactory):
+    def get_orders_with_multiple_sub_orders(self):
```
[the remark this answers is not in the document]
Comment (ghada-37) [reply api_testing/apis/orders_api.py:218]: done

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 4]
```diff
@@ -2118,3 +2118,34 @@ def test033_Duplicate_main_order_and_cahange_materiel_type(self):
...
+        order = self.orders_api.get_orders_with_multiple_sub_orders()
+        self.order_page.search(order['orderNo'])
+        # archive first suborder
```
[the remark this answers is not in the document]
Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test06_orders.py:2132]: done

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 5]
```diff
@@ -2118,3 +2118,34 @@ def test033_Duplicate_main_order_and_cahange_materiel_type(self):
...
+        suborder_data = self.order_page.get_child_table_data()[0]
+        self.order_page.archive_sub_order_from_active_table()
+        # delete it
```
[the remark this answers is not in the document]
Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test06_orders.py:2135]: done

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 6]
```diff
@@ -2118,3 +2118,34 @@ def test033_Duplicate_main_order_and_cahange_materiel_type(self):
...
+        self.order_page.get_archived_items()
+        row = self.order_page.search(order['orderNo'])
+        self.order_page.click_check_box(row[0])
+        self.order_page.delete_selected_item()
+        validation_result = self.base_selenium.get_text(element='orders:confirm_delete')
+        self.assertIn('Please note that the reports of this order will be deleted', validation_result)
+        self.order_page.confirm_popup()
+        # go to analysis page to make sure that analysis reated to that suborder is deleted
```
[the remark this answers is not in the document]
Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test06_orders.py:2143]: done

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 7]
```diff
@@ -2118,3 +2118,34 @@ def test033_Duplicate_main_order_and_cahange_materiel_type(self):
...
+        order = self.orders_api.get_orders_with_multiple_sub_orders()
+        self.order_page.search(order['orderNo'])
+        # archive first suborder
+        suborder_data = self.order_page.get_child_table_data()[0]
+        self.order_page.archive_sub_order_from_active_table()
+        # delete it
+        self.order_page.get_archived_items()
+        row = self.order_page.search(order['orderNo'])
+        self.order_page.click_check_box(row[0])
+        self.order_page.delete_selected_item()
```
[the remark this answers is not in the document]
Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test06_orders.py:2139]: ok

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 8]
```diff
@@ -2118,3 +2118,34 @@ def test033_Duplicate_main_order_and_cahange_materiel_type(self):
...
+        validation_result = self.base_selenium.get_text(element='orders:confirm_delete')
+        self.assertIn('Please note that the reports of this order will be deleted', validation_result)
```
[the remark this answers is not in the document]
Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test06_orders.py]: removed

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 9]
```diff
@@ -2118,3 +2118,34 @@ def test033_Duplicate_main_order_and_cahange_materiel_type(self):
...
+        self.order_page.confirm_popup()
+        # go to analysis page to make sure that analysis reated to that suborder is deleted
+        self.info('Navigate to analysis page to make sure that analysis related to that suborder is deleted')
+        self.order_page.get_orders_page()
+        self.order_page.navigate_to_analysis_tab()
+        self.order_page.sleep_small()
```
[the remark this answers is not in the document]
Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test06_orders.py]: done

File: ui_testing/testcases/basic_tests/test06_orders.py (hunk 1 above)
Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test06_orders.py:2140]: done

**Pull request merged**